### ISSUE TITLE
Use a `lang` JSON property for same-named constructors, instead of naming conventions

### ISSUE THREAD

**Issue by saser** [saser/cakeml#16]
We (currently) have two presLang constructors -- `ModCon` and `ConCon` -- for the two different `Con` constructors from modLang and conLang, respectively. When converting from presLang to JSON the object either has
```
{
    "cons": "Con-modLang"
    ...
}
```
or
```
{
    "cons": "Con-conLang"
    ...
}
```
respectively.

Instead, we should give it a `lang` property, resulting in
```
{
    "cons": "Con",
    "lang": "modLang"
    ...
}
```
or
```
{
    "cons": "Con",
    "lang": "conLang"
    ...
}
```
respectively.

**Comment by hjorthjort** [comment]
Will be fixed by API change

**Comment by hjorthjort** [comment]
Closed by #22
